app/guid-file/controller.ts

```diff
@@ -1,5 +1,278 @@
+    }
+
+    @computed('currentUser.currentUserId', 'user.id')
+    get canEdit(): boolean {
+        const modelUserId = this.user.id;
+
+        return !!modelUserId && modelUserId === this.currentUser.currentUserId;
+    }
+
+    @computed('revision', 'file.currentVersion')
+    get mfrVersion(): number {
+        return this.revision || this.file.currentVersion;
+    }
+
+    // TODO: get this from the model
+    @computed('downloadLink', 'file.currentVersion')
+    get fileVersions(): Promise<any> {
+        return (async () => {
+            const { data } = await $.getJSON(`${this.downloadLink}?revisions=&`);
+            return data;
+        })();
+    }
+
+    @computed('file.name')
+    get isEditableFile(): boolean {
+        const filename = this.file.name;
+        const mimeType = mime.lookup(filename);
+        return !!mimeType && /^text\//.test(mimeType);
+    }
+
+    @computed('file.currentVersion')
+    get fileText() {
+        return Boolean(this.file) && this.file.getContents();
+    }
+
+
+    @restartableTask
+    @waitFor
+    async updateFilter(filter: string) {
+        await timeout(250);
+        this.setProperties({ filter });
+        this.analytics.track('list', 'filter', 'Quick Files - Filter file browser');
+    }
+
+    @computed('allFiles.[]', 'filter', 'sort')
+    get files() {
+        let results: File[] = this.allFiles;
+
+        if (this.filter) {
+            const filterLowerCase = this.filter.toLowerCase();
+            results = results.filter(file => file.name.toLowerCase().includes(filterLowerCase));
+        }
+
+        if (this.sort) {
+            const reverse = this.sort.slice(0, 1) === '-';
+
+            results = A(results).sortBy(this.sort.slice(+reverse));
+
+            if (reverse) {
+                results = results.reverse();
+            }
+        }
+
+        return results;
+    }
+
+    @action
+    download(version: number) {
+        // To do: make this a link that looks like a button rather than calling this
+        const url = `${this.downloadLink}?revision=${version}`;
+        window.location.href = url;
+    }
+
+    @action
+    changeView(button: string) {
+        const show = lookupTable[this.show][button];
+
+        if (show) {
+            this.set('show', show);
+        }
+    }
+
+    @action
+    async save(text: string) {
+        this.analytics.click('button', 'Quick Files - Save');
+
+        try {
+            await this.file.updateContents(text);
+            return this.toast.success(this.intl.t('file_detail.save_success'));
+        } catch (e) {
+            const errorMessage = this.intl.t('file_detail.save_fail');
+            captureException(e, { errorMessage });
+            return this.toast.error(getApiErrorMessage(e), errorMessage);
+        }
+    }
+
+    @action
+    async openFile() {
+        this.set('revision', null);
+    }
+
+    @action
+    versionChange(version: number) {
+        this.set('revision', version);
+    }
+
```

Comment:
I think most of the green lines from the top of the file to here could be removed without affecting your PR. Could you trim that?

Reply:
This is now updated, thank you Brian.